cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/cube/DefaultCube.java

```diff
@@ -154,11 +169,16 @@ SELECT count('read.ops')                                           << measure na
     } else {
       agg = findAggregation(query);
       if (agg == null) {
+        incrementMetric("cube.query.request.failure.count", 1);
         throw new IllegalArgumentException("There's no data aggregated for specified dimensions " +
                                              "to satisfy the query: " + query.toString());
       }
     }
 
+    // tell how many queries end up querying specific pre-aggregated views and resolutions
+    incrementMetric("cube.query.agg." + agg + ".count", 1);
```

Comment:
how does this metric name look like ? , i don't see toString in DefaultAggregation